Aardvark/ARKEmailBugReporter.m

```diff
@@ -342,155 +336,128 @@ - (NSString *)formattedLogMessagesAttachmentExtension;
-        [self _createBugReportWithTitle:textfield.text];
-    }]];
-    
-    [alertController addAction:[UIAlertAction actionWithTitle:cancelButtonTitle style:UIAlertActionStyleDefault handler:NULL]];
-    
-    [alertController addTextFieldWithConfigurationHandler:^(UITextField *textField) {
-        [self _configureAlertTextfield:textField];
+    id <ARKEmailBugReporterPromptingDelegate> const promptPresenter = (self.promptingDelegate ?: [ARKDefaultPromptPresenter new]);
+    [promptPresenter showBugReportingPromptForConfiguration:[self _configurationWithCurrentSettings] completion:^(ARKEmailBugReportConfiguration * _Nullable configuration) {
+        if (configuration != nil) {
+            [self _createBugReportWithConfiguration:configuration];
+        }
     }];
-    
-    UIViewController *viewControllerToPresentAlertController = [UIApplication sharedApplication].keyWindow.rootViewController;
-    while (viewControllerToPresentAlertController.presentedViewController != nil) {
-        viewControllerToPresentAlertController = viewControllerToPresentAlertController.presentedViewController;
-    }
-
-    /*
-     Disabling animations here to avoid potential crashes resulting from unexpected view state in UIKit
-     */
-    [viewControllerToPresentAlertController presentViewController:alertController animated:NO completion:NULL];
-
 }
 
-- (void)_configureAlertTextfield:(UITextField *)textField
+- (ARKEmailBugReportConfiguration *)_configurationWithCurrentSettings;
 {
-    textField.autocapitalizationType = UITextAutocapitalizationTypeSentences;
-    textField.autocorrectionType = UITextAutocorrectionTypeYes;
-    textField.spellCheckingType = UITextSpellCheckingTypeYes;
-    textField.returnKeyType = UIReturnKeyDone;
-}
-
-- (void)_createBugReportWithTitle:(NSString *)title;
-{
-    NSArray *logStores;
+    ARKEmailBugReportConfiguration *const configuration = [[ARKEmailBugReportConfiguration alloc] init];
+    
     if (self.emailAttachmentAdditionsDelegate != nil) {
         NSMutableArray *const filteredLogStores = [NSMutableArray arrayWithCapacity:self.logStores.count];
         for (ARKLogStore *logStore in self.logStores) {
             if ([self.emailAttachmentAdditionsDelegate emailBugReporter:self shouldIncludeLogStoreInBugReport:logStore]) {
                 [filteredLogStores addObject:logStore];
             }
         }
-        logStores = filteredLogStores;
+        configuration.logStores = filteredLogStores;
+        
+        configuration.additionalAttachments = [self.emailAttachmentAdditionsDelegate additionalEmailAttachmentsForEmailBugReporter:self];
+        
     } else {
-        logStores = [self.logStores copy];
+        configuration.logStores = [self.logStores copy];
     }
     
+    configuration.includesScreenshot = self.attachScreenshotToNextBugReport;
+    configuration.includesViewHierarchyDescription = (self.attachScreenshotToNextBugReport && self.attachesViewHierarchyDescriptionWithScreenshot);
```

Comment:
Why is `includesViewHierarchyDescription` dependent on `attachScreenshotToNextBugReport`? Is this behavior documented?

(would also be great to add a test for this)

Reply:
It probably doesn't need to be dependent, but this matches the current behavior.  I'll open a separate issue to remove that dependency.

```
/// If this bug report includes a screenshot, also attach a description of the view hierarchy. Defaults to YES.
@property (nonatomic) BOOL attachesViewHierarchyDescriptionWithScreenshot;
```

Reply:
https://github.com/square/Aardvark/issues/66